snapshot/db.go

```diff
@@ -0,0 +1,50 @@
+package snapshot
+
+import (
+	"github.com/scootdev/scoot/snapshot/git/repo"
+)
+
+// ID identifies a value in DB. Opaque to the client.
+type ID string
+
+// DB is the Scoot Database, allowing creation, distribution, and export of Values.
+type DB interface {
+	// Ingest
+
+	// IngestDir ingests a directory directly.
+	// The created value is a Snapshot.
+	IngestDir(dir string) (ID, error)
+
+	// IngestGitCommit ingests the commit identified by commitish from ingestRepo
+	// commitish may be any string that identifies a commit
+	// The created value is a SnapshotWithHistory.
+	IngestGitCommit(ingestRepo *repo.Repository, commitish string) (ID, error)
+
+	// Operations
+
+	// UnwrapSnapshotHistory unwraps a SnapshotWithHistory and returns a Snapshot ID.
+	// Errors if id does not identify a SnapshotWithHistory.
```

Comment:
I figured it's easier to be strict from the beginning and loosen than vice versa.